src/layouts/DefaultLayout/DefaultLayout.tsx

```diff
@@ -1,54 +1,132 @@
 
-  const [sidebarClasses, setSidebarClasses] = useState<string[]>([]);
-  const [containerClasses, setContainerClasses] = useState<string[]>(isSidebarVisible ? ['with-sidebar'] : []);
+  const [desktopSidebarSx, setDesktopSidebarSx] = useState<SystemStyleObject>({});
+  const [desktopContainerSx, setDesktopContainerSx] =
+    useState<SystemStyleObject>(isDesktopSidebarVisible ? SIDEBAR_VISIBLE_CONTENT_SX : {});
 
-  const showSidebar = useCallback(() => {
-    setSidebarClasses(['sidebar-hidden']);
-    setContainerClasses(['with-sidebar']);
-    setSidebarVisibility(true);
-    setTimeout(() => setSidebarClasses(['sidebar-visible']), 0);
-    setTimeout(() => setSidebarClasses([]), 400);
+  const handleDesktopSidebarShow = useCallback(() => {
+    setDesktopSidebarSx(SIDEBAR_VISIBLE_HEADER_SX);
+    setDesktopContainerSx(SIDEBAR_VISIBLE_CONTENT_SX);
+    setDesktopSidebarVisibility(true);
+    setDesktopHeaderButtonVisibility(false);
+    setTimeout(() => setDesktopSidebarSx({}), 0);
   }, []);
 
-  const hideSidebar = useCallback(() => {
-    setSidebarClasses(['sidebar-visible']);
-    setTimeout(() => setSidebarClasses(['sidebar-hidden']), 0);
-    setContainerClasses([]);
+  const handleDesktopSidebarHide = useCallback(() => {
+    setDesktopContainerSx({});
+    setDesktopHeaderButtonVisibility(true);
+    setTimeout(() => setDesktopSidebarSx(SIDEBAR_VISIBLE_HEADER_SX), 0);
     setTimeout(() => {
-      setSidebarVisibility(false);
-      setSidebarClasses([]);
+      setDesktopSidebarVisibility(false);
+      setDesktopSidebarSx({});
     }, 400);
   }, []);
 
+  const isMobile = useIsMobile();
+  const {
+    isOpen: isMobileSidebarOpen,
+    onOpen: handleMobileSidebarOpen,
+    onClose: handleMobileSidebarClose,
+  } = useDisclosure();
+
+  const isHeaderButtonVisible = useMemo(() => (isMobile ? true : isDesktopHeaderButtonVisible), [isMobile, isDesktopHeaderButtonVisible]);
+  const handleSidebarShow = useCallback(isMobile ? handleMobileSidebarOpen : handleDesktopSidebarShow, [isMobile]);
+  const contentSx = useMemo(() => (isMobile ? {} : desktopContainerSx), [isMobile, desktopContainerSx]);
+
+  /** Returns mobile or desktop sidebar. */
+  function getSidebar(): ReactNode {
+    if (isMobile) {
+      return (
+        <Drawer
+          isOpen={isMobileSidebarOpen}
+          placement="left"
+          onClose={handleMobileSidebarClose}
+        >
+          <DrawerOverlay />
+          <DrawerContent>
+            <Sidebar onSidebarHide={handleMobileSidebarClose} />
+          </DrawerContent>
+        </Drawer>
+      );
+    }
+    return isDesktopSidebarVisible && (
+      <Box
+        position="fixed"
+        width={60}
+        transition="0.3s ease-in-out"
+        transform="translateX(0)"
+        sx={desktopSidebarSx}
+      >
+        <Sidebar onSidebarHide={handleDesktopSidebarHide} />
+      </Box>
+    );
+  }
+
+  const isFloatingHeaderVisible = useFloatingHeader();
```

Comment:
Usually, we try to declare hooks first and only then other constants and functions